router/core/request_context_fields.go

```diff
@@ -32,6 +34,23 @@ const (
 
 // Helper functions to create zap fields for custom attributes.
 
+func NewExpressionLogField(val any, key string, defaultValue any) zap.Field {
+	// Depending on the condition exprlang will dereference reportError sometimes
+	// and as the method receivers are all pointers (*reportError)
+	// the Error() function wont get called unless its of type *reportError
+	switch result := val.(type) {
+	case reportError:
```

Comment:
As the comment mentions, depending on the expression `exprlang` will return `reportError` instead of the pointer version `*reportError`, which means that Error() wont end up being called

Reply:
I can't follow. `reportError` is an internal error from our router and has nothing to do with `expr`. Could you eloborate?

Reply:
Now I understand. You were referring to the returned error of the expression. Why do you believe it is important toi align this. The value should not matter because we will pass it straight to the logger which accepts any.

Reply:
Even though the logger accepts any, the final log message ends up becoming a memory location. This becomes a problem when a customer expects the error message to be printed.

Reply:
I'd expect that whatever the user returns in an expression that zap is able to log it and that the pointer is dereferenced automatically before printing.

Reply:
From my testing so far no, but let me double check a bit more and get back to you.